src/main/java/br/com/quintoandar/javajwt/QuintoAndarKeycloakPublicKeyService.java

```diff
@@ -0,0 +1,49 @@
+package br.com.quintoandar.javajwt;
+
+import br.com.quintoandar.javajwt.config.QuintoandarProperties;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.net.URLConnection;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+
+public class QuintoAndarKeycloakPublicKeyService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(QuintoAndarKeycloakPublicKeyService.class);
+
+    private static final int BUFFER_SIZE = 8192;
+
+    private final QuintoandarProperties quintoandarProperties;
+
+    @Autowired
+    public QuintoAndarKeycloakPublicKeyService(final QuintoandarProperties quintoandarProperties) {
+        this.quintoandarProperties = quintoandarProperties;
+    }
+
+    // visible for testing
+    protected String fetchKeycloakPublicKey() throws IOException {
```

Comment:
shouldn't this method be public?

Reply:
Hum... Shouldn't protected be enough? Since I'm using it entirely inside the javajwt package and it's only used in the package itself. Do you think there is a case where we might want to call this method from the repo using this package?

Reply:
hmm makes sense of being protected, but in this case, it isn't protected just for testing right? but to be visible only in the package.

Reply:
Oh, good point, the comment is misleading! Will remove it after checking the other things you commented here ^^

Reply:
Done!